File: day2/selection.java
package day2;
// Write a function to perform a binary search on a sorted array of integers to find a target element.
class Sorting {
    public static void printArray(int arr[]) {
        for(int i=0; i<arr.length; i++) {
            System.out.print(arr[i]+" ");
        }
        System.out.println();
    }
 
 
    public static void main(String args[]) {
        int arr[] = {5, 3, 8, 2, 7, 1, 6, 4};
 
 
        //selection sort
        for(int i=0; i<arr.length-1; i++) {
            int smallest = i;
            for(int j=i+1; j<arr.length; j++) {
                if(arr[j] < arr[smallest]) {
                    smallest = j;
                }
            }
            //swap
            int temp = arr[smallest];
            arr[smallest] = arr[i];
            arr[i] = temp;
        }
 
 
        printArray(arr);
    }
} 

File: day2/selection.cpp
// Problem Statement:
// Implement the selection sort algorithm to sort an array of integers.

// Input:
// An array of integers.

// Output:
// The sorted array.

// Example:
// Input:
// [5, 3, 8, 2, 7, 1, 6, 4]
// Output:
// [1, 2, 3, 4, 5, 6, 7, 8]

#include <iostream>
using namespace std;

class Sorting {
public:
    static void printArray(int arr[], int size) {
        for(int i = 0; i < size; i++) {
            cout << arr[i] << " ";
        }
        cout << endl;
    }
 
    static void selectionSort(int arr[], int size) {
        for(int i = 0; i < size - 1; i++) {
            int smallest = i;
            for(int j = i + 1; j < size; j++) {
                if(arr[j] < arr[smallest]) {
                    smallest = j;
                }
            }
            //swap
            int temp = arr[smallest];
            arr[smallest] = arr[i];
            arr[i] = temp;
        }
    }
};

int main() {
    int arr[] = {5, 3, 8, 2, 7, 1, 6, 4};
    int size = sizeof(arr) / sizeof(arr[0]);
 
    Sorting::selectionSort(arr, size);
    Sorting::printArray(arr, size);

    return 0;
}
